enterprise/server/remote_execution/container/container.go

```diff
@@ -437,11 +437,15 @@ type VM interface {
 // PullImageIfNecessary pulls the image configured for the container if it
 // is not cached locally.
 func PullImageIfNecessary(ctx context.Context, env environment.Env, ctr CommandContainer, creds oci.Credentials, imageRef string) error {
-	ctx, span := tracing.StartSpan(ctx)
-	defer span.End()
-	if *debugUseLocalImagesOnly {
+	if *debugUseLocalImagesOnly || imageRef == "" {
```

Comment:
👀

Reply:
This is the only thing that seems like it could even take time... I'm suspicious

Reply:
It is for the bare runner, which we often use for mac executors

Reply:
gah sorry, i thought this was in podman.go, my bad